integration_test/arcilator/JIT/reg.mlir

```diff
@@ -41,6 +41,7 @@ func.func @main() {
   %step = arith.constant 1 : index
 
   arc.sim.instantiate @counter as %model {
+    arc.sim.step %model : !arc.sim.instance<@counter>
```

Comment:
Is this change required to run initialization?

Reply:
Yeah. In its current form, a call to the `initial` function only executes `seq.initial` blocks and initializes the storage for states. It does not propagate those changes to outputs and side-effecting ops though. I wasn't sure if the `initial` function should also call `eval` once after initialization (potentially triggering first state updates and other things to run immediately), or if the user might want more fine grained control over when the first eval happens. WDYT?

Reply:
Yeah I see that makes sense. I agree that we want to avoid implicitly calling `eval` from `initial`